tests/kernel/interrupt/src/dynamic_isr.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+extern void z_irq_spurious(void *unused);
+
+/* Simple whitebox test of dynamic interrupt handlers as implemented
+ * by the GEN_SW_ISR feature.
+ */
+#if (defined(CONFIG_DYNAMIC_INTERRUPTS)		\
+	&& defined(CONFIG_GEN_SW_ISR_TABLE))
+#define DYNTEST 1
```

Comment:
@stephanosio i wonder if you could clean up this local macro define - and use directly the corresponding CONFIG_ macro defines.

Reply:
Sure, I will address that in the PR for https://github.com/zephyrproject-rtos/zephyr/pull/23614#discussion_r399184649 as well.